### ISSUE TITLE
Call to removeItem unhide hidden items.

### ISSUE THREAD

**Issue by msmttchr** [pyqtgraph/pyqtgraph#2066]
### Short description
If you run the example attached, and click first on `Hide` and then on `Remove`, you will end up with a graph showing only the curve whose value is 10, but this curve should actually be hidden.

### Code to reproduce

```python
from pyqtgraph.Qt import QtGui, QtCore
import numpy as np
import pyqtgraph as pg

app = pg.mkQApp()
mw = QtGui.QMainWindow()
cw = QtGui.QWidget()
mw.setCentralWidget(cw)
l = QtGui.QVBoxLayout()
cw.setLayout(l)


button_hide = QtGui.QPushButton("Hide")
button_remove = QtGui.QPushButton("Remove")
pw = pg.PlotWidget(name='Plot1') 
l.addWidget(pw)
l.addWidget(button_hide)
l.addWidget(button_remove)

mw.show()

plots=[]
plots.append(pw.plot(y=[10,]*2, x=list(range(2))))
plots.append(pw.plot(y=[20,]*2, x=list(range(2))))

button_hide.clicked.connect(lambda x: plots[0].hide())
button_remove.clicked.connect(lambda x: pw.removeItem(plots[1]))

if __name__ == '__main__':
    pg.exec()
```

### Expected behavior

No curve should be shown since the curve 20 is removed and the curve 10 is hidden.
### Real behavior

Curve 10 is shown.

### Tested environment(s)

 * PyQtGraph version: 0.12.2
 * Qt Python binding: 
 * Python version: PyQt5 5.15.4 Qt 5.15.2
 * NumPy version: 1.20.1
 * Operating system: Windows 10
 * Installation method: pip

**Comment by sem-geologist** [comment]
try pyqtgraph 0.12.3 or master, I believe this was fixed in PR #1973

**Comment by msmttchr** [comment]
Indeed, version 0.12.3 fixed the problem. Thanks
